src/test/java/ru/yandex/practicum/filmorate/controller/ControllerTest.java

```diff
@@ -0,0 +1,55 @@
+package ru.yandex.practicum.filmorate.controller;
+
+import org.junit.jupiter.api.Test;
+import ru.yandex.practicum.filmorate.model.Entity;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public abstract class ControllerTest<T extends Entity> {
+
+    private final BasicController<T>  controller;
+
+    protected ControllerTest(BasicController<T> controller) {
+        this.controller = controller;
+    }
+
+    @Test()
+    void addNewNull() {
+        assertThrows(Exception.class, () -> controller.addNew(null));
+    }
+
+    @Test
+    void updateNull() {
+        assertThrows(NullPointerException.class, () -> controller.update(null));
+    }
+
+    void getAll(T original) {
+        assertEquals(0, controller.getAll().size());
```

Comment:
Можно упростить проверку до:
```suggestion
          assertTrue(controller.getAll().isEmpty());
```
Метод `isEmpty()` возвращает `true`, если `размер списка == 0` или `false` в любом другом случае

Reply:
да так проще. действительно.
просто я встречал ситуации ,когда IDEA часто предлагала assertEquals() вместо assertTrue()
сейчас я наверное понял почему. кажется она настолько умна, чтобы предусматривать, что будет легче читаться.
assertTrue(controller.getAll().isEmpty()) здесь получается легче человеко-читаемо. буду проверять дальше эту теорию